linkedlist/src/main/java/ua/hillel/LinkedList.java

```diff
@@ -1,69 +1,162 @@
 package ua.hillel;
 
+import java.util.Objects;
+
 /**
  * Implement doubly linked list that is based on {@link Node<T>} class (you should implement it as well).
  * This is a simplified version of {@link java.util.LinkedList}.
  */
 public class LinkedList<T> implements List<T> {
-
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
 
     @Override
     public void add(T element) {
-
+        Node<T> node = new Node<>(element);
+        if (last == null) {
+            first = node;
+        } else {
+            last.next = node;
+            node.previous = last;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (index > size || index < 0) {
+            return;
+        }
+        Node<T> node = new Node<>(element);
+        if (index == size) {
+            add(element);
+            return;
+        }
+        if (index == 0) {
+            first.previous = node;
+            node.next = first;
+            first = node;
+            size++;
+            return;
+        }
+        Node<T> currentNode = getNode(index);
+        node.previous = currentNode.previous;
+        node.next = currentNode;
+        currentNode.previous.next = node;
+        currentNode.previous = node;
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            return null;
+        }
+        return getNode(index).element;
     }
 
     @Override
     public T get(T element) {
+        Node<T> node = first;
+        while (node != null) {
+            if (Objects.equals(node.element, element)) {
+                return node.element;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public T getFirst() {
-        return null;
+        return first.element;
     }
 
     @Override
     public T getLast() {
-        return null;
+        return last.element;
```

Comment:
What if `last` == null? You'll get NullPointerException. Consider adding some checks.